### PR TITLE
Clang warning fixes

### PR DESCRIPTION
**Pull request by TheComputerGuy96**
This should be enough to finally enable `-Werror` in non-MSVC cases (MSVC uses a `/WX` flag for that instead) :frog: 

For the zlib issue I could've just fixed the problematic prototype part but updating zlib probably brings some security fixes too so I think that's a better option

### CODE REVIEW

File: src/main/formats/CPS/CPS2Scanner.cpp [hunk 1]
```diff
@@ -104,6 +104,8 @@ void CPS2Scanner::Scan(RawFile *file, void *info) {
       if (fmt_ver >= VER_130 && !seqRomGroupEntry->GetHexAttribute("artic_table", &artic_table_offset))
         return;
       break;
+    default:
+      break;
   }
```
Comment (mikelow) [inline src/main/formats/CPS/CPS2Scanner.cpp:109]: Instead of default: break; here, we should handle the missing enum cases. In the event of VER_UNDEFINED, we should log an error and return;

File: src/main/formats/CPS/CPS1Scanner.cpp [hunk 2]
```diff
@@ -103,6 +103,8 @@ void CPS1Scanner::LoadCPS1(MAMEGame *gameentry, CPSFormatVer fmt_ver) {
     case VER_CPS1_502:
       seq_table_length = (uint32_t) (programFile->GetShortBE(seq_table_offset) * 2) + ptrsStart;
       break;
+    default:
+      break;
   }
```
Comment (mikelow) [inline src/main/formats/CPS/CPS1Scanner.cpp:108]: This is an error state - the version is non-CPS1 or undefined. We should log an error and return.

File: src/main/formats/MoriSnes/MoriSnesSeq.cpp [hunk 3]
```diff
@@ -732,7 +732,7 @@ void MoriSnesTrack::ParseInstrumentEvents(uint16_t offset, uint8_t instrNum, boo
 
   while (bContinue) {
     uint16_t beginOffset = curOffset;
-    if (curOffset >= 0x10000) {
```
Comment (mikelow) [inline src/main/formats/MoriSnes/MoriSnesSeq.cpp:735]: Why this change?

File: src/main/formats/MoriSnes/MoriSnesSeq.cpp [hunk 4]
```diff
@@ -762,7 +762,7 @@ void MoriSnesTrack::ParseInstrumentEvents(uint16_t offset, uint8_t instrNum, boo
       }
 
       beginOffset = curOffset;
-      if (curOffset >= 0x10000) {
```
Comment (mikelow) [inline src/main/formats/MoriSnes/MoriSnesSeq.cpp:765]: Ditto, why are you changing the value?

**Comment by mikelow** [comment]
Did you do any testing? You made a number of logic changes in format code.

File: src/main/formats/FFT/FFTSeq.cpp [hunk 5]
```diff
@@ -188,9 +188,11 @@ bool FFTTrack::ReadEvent(void) {
         //Octave Event
 
         //Set octave
-      case 0x94:
-        AddSetOctave(curOffset - 2, 2, GetByte(curOffset++));
+      case 0x94: {
+        uint8_t newOctave = GetByte(curOffset++);
+        AddSetOctave(curOffset - 2, 2, newOctave);
```
Comment (mikelow) [inline src/main/formats/FFT/FFTSeq.cpp:193]: This should be:
`AddSetOctave(beginOffset, 2, newOctave);`

Using curOffset was probably a dumb mistake of mine.

File: src/main/formats/FFT/FFTSeq.cpp [hunk 6]
```diff
@@ -477,9 +479,11 @@ bool FFTTrack::ReadEvent(void) {
         //Volume Event
 
         // Volume
-      case 0xE0:
-        AddVol(curOffset - 2, 2, GetByte(curOffset++));
+      case 0xE0: {
+        uint8_t volByte = GetByte(curOffset++);
+        AddVol(curOffset - 2, 2, volByte);
```
Comment (mikelow) [inline src/main/formats/FFT/FFTSeq.cpp:484]: Likewise, should be:
AddVol(beginOffset, curOffset - beginOffset, volByte);

File: src/main/formats/FFT/FFTSeq.cpp [hunk 7]
```diff
@@ -528,9 +532,11 @@ bool FFTTrack::ReadEvent(void) {
         //Panpot Event
 
         // Panpot
-      case 0xE8:
-        AddPan(curOffset - 2, 2, GetByte(curOffset++));
+      case 0xE8: {
+        uint8_t panpotByte = GetByte(curOffset++);
+        AddPan(curOffset - 2, 2, panpotByte);
```
Comment (mikelow) [inline src/main/formats/FFT/FFTSeq.cpp:537]: AddPan(beginOffset, curOffset - beginOffset, panpotByte);

File: src/main/formats/KonamiSnes/KonamiSnesSeq.cpp [hunk 8]
```diff
@@ -234,6 +234,9 @@ void KonamiSnesSeq::LoadEventMap() {
       EventMap[0xfb] = EVENT_ADSR2;
       EventMap[0xfc] = EVENT_PROGCHANGEVOL;
       break;
+
+    default:
+      break;
   }
```
Comment (mikelow) [inline src/main/formats/KonamiSnes/KonamiSnesSeq.cpp:240]: I think we should put a warning about the format version being unknown:
`L_WARN("Unknown version of Konami SNES format.");`

**Comment by mikelow** [comment]
@TheComputerGuy96 from where did you get this version of zlib?
@sykhro should we make zlib a git submodule given that we're doing that for spdlog?

**Comment by sykhro** [comment]
> @TheComputerGuy96 from where did you get this version of zlib? @sykhro should we make zlib a git submodule given that we're doing that for spdlog?

Yes, at that point we could even switch to https://github.com/zlib-ng/zlib-ng/

**Comment by mikelow** [comment]
@TheComputerGuy96 I'm happy to make a new branch that brings in your commits sans the vendored zlib, fixes the issues I mentioned, and introduces zlib-ng via git submodule.

**Comment by TheComputerGuy96** [comment]
> from where did you get this version of zlib?

I unpacked the zlib 1.3.1 tarball (specifically https://github.com/madler/zlib/releases/download/v1.3.1/zlib-1.3.1.tar.gz) and started removing files from it to reduce the space of the module and modified CMakeLists,txt to remove the shared library stuff and build the minizip library together too

> At that point we could even switch to https://github.com/zlib-ng/zlib-ng/

I'm fine with this because Arch has packaged both zlib-ng and minizip-ng (so a submodule won't be needed there)

File: src/main/formats/MoriSnes/MoriSnesSeq.cpp (hunk 4 above)
Comment (TheComputerGuy96) [reply src/main/formats/MoriSnes/MoriSnesSeq.cpp:765]: Because Clang was complaining about that (integer overflow?) check being always false

**Comment by TheComputerGuy96** [comment]
> Did you do any testing? You made a number of logic changes in format code.

I would have to get music of multiple different formats to test this PR properly

**Comment by mikelow** [comment]
Closing this now that it has been integrated into #477 and merged. 

I am working on replacing zlib/minizip with zlib-ng/minizip-ng and including them as git submodules. I will put up that PR when it is ready.
